### PR TITLE
Fix #17

### LINKED ISSUE #17: Implement model for database and it's functions

### PR DESCRIPTION
**Pull request by Valentin889**
This is a default implementation for view model and database use with async functions.
There is a strong default structure for creating new screen, it needs strong review as it will be a base for interacting with the db and creating new screens

### CODE REVIEW

File: app/src/main/java/com/android/partagix/model/loan/loan.kt
Comment (LozanoCyrielle) [inline app/src/main/java/com/android/partagix/model/loan/loan.kt:1]: A loan is only one item? can we use a list of items for a loan ?

File: app/src/main/java/com/android/partagix/model/item/item.kt
Comment (LozanoCyrielle) [inline app/src/main/java/com/android/partagix/model/item/item.kt:1]: maybe add object name, description, quantity, visibility, disponibility, location (and the other infos we would want to be able to access for each item)

File: app/src/main/java/com/android/partagix/model/user/user.kt
Comment (LozanoCyrielle) [inline app/src/main/java/com/android/partagix/model/user/user.kt:1]: Add the comment about a user too?

File: app/src/main/java/com/android/partagix/model/Database.kt [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            }
+            onSuccess(ret)
+        }.addOnFailureListener() {
+            println("----- error $it")
+        }
+    }
+
+
+    fun getNewUid(): Long {
+        newId += 1
```
Comment (LozanoCyrielle) [inline app/src/main/java/com/android/partagix/model/Database.kt]: Not really safe for the id
maybe use something like : 
```
val uidDocument = collection.document()
return uidDocument.id
```

File: app/src/main/java/com/android/partagix/ui/App.kt
Comment (LozanoCyrielle) [inline app/src/main/java/com/android/partagix/ui/App.kt:1]: Don't forget to change the route name

File: app/src/main/java/com/android/partagix/ui/navigation/NavigationActions.kt
Comment (LozanoCyrielle) [inline app/src/main/java/com/android/partagix/ui/navigation/NavigationActions.kt:1]: Don't forget to change the destination  and routes, so it goes with our app

File: app/src/main/java/com/android/partagix/ui/screens/Inventory.kt
Comment (LozanoCyrielle) [inline app/src/main/java/com/android/partagix/ui/screens/Inventory.kt:1]: delete the test tag

**Review by LozanoCyrielle — CHANGES_REQUESTED** [review]
Delete the println used for tests
Delete test tags (even if they are relics of the bootcamps :) ) 
The system to get newId isn't really "safe"
The state of the loan (pending, accepted, refused) isn't stored anywhere

File: app/src/main/java/com/android/partagix/model/item/item.kt [hunk 2]
```diff
@@ -0,0 +1 @@
+data class Item(val id: Long, val idCategory: Long)
```
Comment (KassFlute) [inline app/src/main/java/com/android/partagix/model/item/item.kt:1]: No Item name ?
And many other fields ?

Comment (Valentin889) [reply app/src/main/java/com/android/partagix/model/item/item.kt:1]: Yes, je vais add des trucs, l'idée c'était juste de mettre les basics et ensuite n'importe qui peut add tant que c'est fait aux 2 endroits

**Review by skyrag — CHANGES_REQUESTED** [review]
check the commentary left by cyrielle

File: app/src/main/java/com/android/partagix/model/Database.kt
Comment (KassFlute) [inline app/src/main/java/com/android/partagix/model/Database.kt:1]: Il ne faudrait pas dans Database.kt une manière d'obtenir l'inventaire d'un user donné ?
Ou ça va pas dans database.kt ce genre de truc

File: app/src/main/java/com/android/partagix/model/loan/loan.kt
Comment (KassFlute) [inline app/src/main/java/com/android/partagix/model/loan/loan.kt:1]: A loan can't have multiple items ?

File: app/src/main/java/com/android/partagix/model/Database.kt
Comment (Valentin889) [reply app/src/main/java/com/android/partagix/model/Database.kt:1]: Yes, j'ai juste mis des basics fonctions (je viens de rajouter celle là) c'est juste un modèle il faudra rajouter des fonctions dans db

File: app/src/main/java/com/android/partagix/model/loan/loan.kt
Comment (Valentin889) [reply app/src/main/java/com/android/partagix/model/loan/loan.kt:1]: Yes j'ai ajouter une nouvelle table dans la db et je vais écrire la conversion soon

**Review by KassFlute — APPROVED** [review]

File: app/src/main/java/com/android/partagix/model/Database.kt
Comment (KassFlute) [reply app/src/main/java/com/android/partagix/model/Database.kt:1]: oops English
okay it's fine, I just wanted to be sure to understand

**Review by Mass-14 — APPROVED** [review]

**Review by AntoineBut — APPROVED** [review]
Looks good, it could be useful for us to have a more visual overview of the app's structure ro make sure everyone understands how we are supposed to do things !

Thank you for the good work 👍

**Review by Mass-14 — APPROVED** [review]

**Review by skyrag — APPROVED** [review]
all good

**Pull request merged**